boards/arm/b_u585i_iot02a/board.cmake

```
@@ -1,4 +1,13 @@
 # SPDX-License-Identifier: Apache-2.0
+if(CONFIG_BUILD_WITH_TFM)
+  set(TFM_FLASH_BASE_ADDRESS 0x0C000000)
+
+  if (CONFIG_HAS_FLASH_LOAD_OFFSET)
```

This should cause `west flash` to point to the correct .hex file when TF-M is enabled:

```suggestion
  # Flash merged TF-M + Zephyr binary
  set_property(TARGET runners_yaml_props_target PROPERTY hex_file "${CMAKE_BINARY_DIR}/tfm_merged.hex")

  if (CONFIG_HAS_FLASH_LOAD_OFFSET)
```

---

Working great, thanks!
Will update this board and provide the fix for others.